packages/esm-patient-labs-app/src/test-results/filter/filter-set.component.tsx

```diff
@@ -67,7 +67,8 @@ const FilterNodeParent = ({ root, itemNumber }: filterNodeParentProps) => {
   const tablet = useLayoutType() === 'tablet';
   const [expandAll, setExpandAll] = useState<boolean | undefined>(undefined);
 
-  if (!root.subSets) return;
+  // Filter out root nodes without data
+  if (!root.subSets || !root.subSets.some((node) => node.hasData || node.subSets)) return null;
```

Comment:
Does this work for a root that has subSets with subSets, but those grandchildren have no data?

Reply:
Let me do this recursively.